[ROCm] re-enable tensorexpr and test_openmp

arindamroy-eng opened the pull request:
The following tests are being re-enabled for ROCm:
- test_openmp.py
- TestTensorExprPyBind tests in test_tensorexpr_pybind.py

arindamroy-eng committed:
ROCM: Set proper LLVM path for RCOM builds

By setting this path, TestTensorExprPyBind
tests in test_tensorexpr_pybind.py will be
enabled for ROCM

Signed-off-by: Arindam Roy <rarindam@gmail.com>

arindamroy-eng committed:
ROCM: Renable test_openmp for ROCM

Signed-off-by: Arindam Roy <rarindam@gmail.com>

arindamroy-eng commented:
@pruthvistony needs merge

pruthvistony commented:
@arindamroy-eng ,
Check if the change is causing a problem for the libtorch_cpu.so build, due to llvm path changes.
What is the result from your local testing?

arindamroy-eng commented:
find /opt/ -name libtorch_cpu.so

/opt/conda/lib/python3.7/site-packages/torch/lib/libtorch_cpu.so



The test_tensorexpr runs fine. It actually the test_mobile_nnc, which not a python file, has problem finding the libtorch_cpu.so as its not in its default LD path.

echo $LD_LIBRARY_PATH

/opt/ompi/lib:/opt/rocm/lib:/usr/local/lib

export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/conda/lib/python3.7/site-packages/torch/lib/

/opt/conda/lib/python3.7/site-packages/torch/bin/test_mobile_nnc --gtest_output=xml:/test_mobile_nnc.xml
Note: Google Test filter = *-*_CUDA:*_MultiCUDA

[==========] Running 6 tests from 2 test suites.

[----------] Global test environment set-up.

arindamroy-eng committed:
ROCM:Add libtorch_cpu.so path for aot_mobile_nnc

When LLVM is being enabled for ROCM, it
seems that the test_mobile_nnc does not
have the libtorch_cpu.so in its LD_LIBRARY_PATH.
Adding this path temorarily to run the binary.

Signed-off-by: Arindam Roy <rarindam@gmail.com>

arindamroy-eng commented:
In the linuixfocal build, the art test is passing, but it is writing its result to /, hence its failing:

+ /opt/conda/lib/python3.7/site-packages/torch/bin/test_mobile_nnc --gtest_output=xml:/test_mobile_nnc.xml
[69041](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69042)
Note: Google Test filter = *-*_CUDA:*_MultiCUDA
[69042](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69043)
[==========] Running 6 tests from 2 test suites.
[69043](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69044)
[----------] Global test environment set-up.
[69044](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69045)
[----------] 4 tests from Function
[69045](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69046)
[ RUN      ] Function.ExecuteSlowMul
[69046](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69047)
[       OK ] Function.ExecuteSlowMul (25 ms)
[69047](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69048)
[ RUN      ] Function.Serialization
[69048](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69049)
[       OK ] Function.Serialization (0 ms)
[69049](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69050)
[ RUN      ] Function.ValidInput
[69050](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69051)
[       OK ] Function.ValidInput (0 ms)
[69051](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69052)
[ RUN      ] Function.InvalidInput
[69052](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69053)
[       OK ] Function.InvalidInput (10 ms)
[69053](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69054)
[----------] 4 tests from Function (35 ms total)
[69054](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69055)
[69055](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69056)
[----------] 2 tests from MobileNNCRegistryTest
[69056](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69057)
[ RUN      ] MobileNNCRegistryTest.FindAndRun
[69057](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69058)
[       OK ] MobileNNCRegistryTest.FindAndRun (0 ms)
[69058](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69059)
[ RUN      ] MobileNNCRegistryTest.NoKernel
[69059](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69060)
[       OK ] MobileNNCRegistryTest.NoKernel (0 ms)
[69060](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69061)
[----------] 2 tests from MobileNNCRegistryTest (0 ms total)
[69061](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69062)
[69062](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69063)
[----------] Global test environment tear-down
[69063](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69064)
[69064](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69065)
[ FATAL ] /var/lib/jenkins/workspace/third_party/googletest/googletest/src/gtest.cc:190:: Unable to open file "/test_mobile_nnc.xml"
[69065](https://github.com/pytorch/pytorch/runs/7400086497?check_suite_focus=true#step:7:69066)
.jenkins/pytorch/test.sh: line 329: 42736 Aborted                 (core dumped) LD_LIBRARY_PATH="$LD_LIBRARY_PATH":/opt/conda/lib/python3.7/site-packages/torch/lib/ "$TORCH_BIN_DIR"/test_mobile_nnc --gtest_output=xml:$TEST_REPORTS_DIR/test_mobile_nnc.xml

arindamroy-eng committed:
ROCM: Disbale aot test for rocm

This test is for mobile device, and is
throwing unrelated errors once llvm is being
enabled.
Hence disabling for ROCM.

Signed-off-by: Arindam Roy <rarindam@gmail.com>

arindamroy-eng commented:
The failing test is in torch dynamo.
test_bce_with_logits_has_correct_forward_grad (__main__.TestNN)
File "/opt/conda/lib/python3.7/site-packages/torch/fx/graph_module.py", line 267, in __call__
[75247](https://github.com/pytorch/pytorch/runs/7509735075?check_suite_focus=true#step:10:75248)
    return super(self.cls, obj).__call__(*args, **kwargs)  # type: ignore[misc]
[75248](https://github.com/pytorch/pytorch/runs/7509735075?check_suite_focus=true#step:10:75249)
  File "/opt/conda/lib/python3.7/site-packages/torch/nn/modules/module.py", line 1186, in _call_impl
[75249](https://github.com/pytorch/pytorch/runs/7509735075?check_suite_focus=true#step:10:75250)
    return forward_call(*input, **kwargs)
[75250](https://github.com/pytorch/pytorch/runs/7509735075?check_suite_focus=true#step:10:75251)
TypeError: forward() takes 2 positional arguments but 3 were given
[75251](https://github.com/pytorch/pytorch/runs/7509735075?check_suite_focus=true#step:10:75252)

This is unrelated to the PR.
@bdhirsh

arindamroy-eng commented:
@pruthvistony @jeffdaily @jithunnair-amd This can be merged now.

jithunnair-amd commented:
@malfet Who would be the right person to review this PR?

jeffdaily reviewed (changes requested):
Please revert the following commits:​
- 60e170a3d0017c4220133eaac396701ece4fa94a
- a4947e55c8a0fcf6f8d20d354e1ae3015f29518b

Then please apply this patch:
```diff
diff --git a/.jenkins/pytorch/test.sh b/.jenkins/pytorch/test.sh
index b476d25250..432656c6af 100755
--- a/.jenkins/pytorch/test.sh
+++ b/.jenkins/pytorch/test.sh
@@ -322,6 +322,9 @@ test_libtorch() {

 test_aot_compilation() {
   echo "Testing Ahead of Time compilation"
+  ln -sf "$TORCH_LIB_DIR"/libc10* "$TORCH_BIN_DIR"
+  ln -sf "$TORCH_LIB_DIR"/libtorch* "$TORCH_BIN_DIR"
+
   if [ -f "$TORCH_BIN_DIR"/test_mobile_nnc ]; then "$TORCH_BIN_DIR"/test_mobile_nnc --gtest_output=xml:$TEST_REPORTS_DIR/test_mobile_nnc.xml; fi
   # shellcheck source=test/mobile/nnc/test_aot_compile.sh
   if [ -f "$TORCH_BIN_DIR"/aot_model_compiler_test ]; then source test/mobile/nnc/test_aot_compile.sh; fi
```

This follows the precedent elsewhere in test.sh to symlink the libs prior to running cpp tests.  

The reason it had failed for rocm was due to the order of tests in the test.sh file.  Targets other than rocm were symlinking as part of their steps, and those other tests were executed prior to `test_aot_compilation()` but rocm skipped them and therefore skipped the symlinking.  And since `test_aot_compilation()` didn't do its own symlinking, it would fail for rocm.

arindamroy-eng committed:
Revert "ROCM: Disbale aot test for rocm"

This reverts commit 60e170a3d0017c4220133eaac396701ece4fa94a.

arindamroy-eng committed:
Revert "ROCM:Add libtorch_cpu.so path for aot_mobile_nnc"

This reverts commit a4947e55c8a0fcf6f8d20d354e1ae3015f29518b.

arindamroy-eng committed:
ROCM: Fix test_aot_compilation() for ROCM

This follows the precedent elsewhere in test.sh
to symlink the libs prior to running cpp tests.

The reason it had failed for rocm was due to the
order of tests in the test.sh file. Targets other
than rocm were symlinking as part of their steps,
and those other tests were executed prior to
test_aot_compilation() but rocm skipped them
and therefore skipped the symlinking. And since
test_aot_compilation() didn't do its own symlinking,
it would fail for rocm.

Signed-off-by: Arindam Roy <rarindam@gmail.com>

jeffdaily commented on .jenkins/pytorch/test.sh:
```diff
@@ -329,6 +329,9 @@ test_libtorch() {
   echo "Testing Ahead of Time compilation"
+  ln -sf "$TORCH_LIB_DIR"/libc10* "$TORCH_BIN_DIR"
+  ln -sf "$TORCH_LIB_DIR"/libtorch* "$TORCH_BIN_DIR"
+
```
The TEST_REPORTS_DIR wasn't set, that's why we had a failure.

```suggestion

  # Make test_reports directory
  # NB: the ending test_libtorch must match the current function name for the current
  # test reporting process (in print_test_stats.py) to function as expected.
  TEST_REPORTS_DIR=test/test-reports/cpp-unittest/test_aot_compilation
  mkdir -p $TEST_REPORTS_DIR
```

jeffdaily committed:
set TEST_REPORTS_DIR for test_aot_compilation()

arindamroy-eng commented:
@pytorchmergebot please merge

malfet commented:
@pytorchbot rebase

malfet commented:
@pytorchbot merge -g
